tools/glb-viewer/src/debug.js

```diff
@@ -0,0 +1,198 @@
+var debugLayerFront = null;
+var debugLayerBack = null;
+
+var DebugLines = function (app, camera) {
+    if (!debugLayerFront) {
+        // construct the debug layer
+        debugLayerFront = new pc.Layer({
+            enabled: true,
+            name: 'Debug Layer',
+            opaqueSortMode: pc.SORTMODE_NONE,
+            transparentSortMode: pc.SORTMODE_NONE,
+            passThrough: true
+        });
+
+        debugLayerBack = new pc.Layer({
+            enabled: true,
+            name: 'Debug Layer Behind',
+            opaqueSortMode: pc.SORTMODE_NONE,
+            transparentSortMode: pc.SORTMODE_NONE,
+            passThrough: true,
+            overrideClear: true,
+            clearDepthBuffer: true
+        });
+
+        app.scene.layers.pushTransparent(debugLayerFront);
+        app.scene.layers.pushTransparent(debugLayerBack);
+        camera.camera.layers = camera.camera.layers.concat([debugLayerFront.id, debugLayerBack.id]);
+    }
+
+    var device = app.graphicsDevice;
+
+    var vertexFormat = new pc.VertexFormat(device, [
+        { semantic: pc.SEMANTIC_POSITION, components: 3, type: pc.TYPE_FLOAT32 },
+        { semantic: pc.SEMANTIC_COLOR, components: 4, type: pc.TYPE_UINT8, normalize: true }
+    ]);
+
+    // construct the mesh
+    var mesh = new pc.Mesh();
+    mesh.vertexBuffer = new pc.VertexBuffer(device, vertexFormat, 1024, pc.BUFFER_DYNAMIC);
+    mesh.primitive[0].type = pc.PRIMITIVE_LINES;
+    mesh.primitive[0].base = 0;
+    mesh.primitive[0].indexed = false;
+    mesh.primitive[0].count = 0;
+
+    // construct the material
+    var material = new pc.BasicMaterial();
+    material.blend = false;
+    material.blendType = pc.BLEND_NORMAL;
```

Comment:
On the previous line, you disable blending. Then, setting `blendType` to `pc.BLEND_NORMAL` will reenable blending. Did you want `pc.BLEND_NONE`? https://github.com/playcanvas/engine/blob/master/src/scene/materials/material.js#L178

Reply:
this is being addressed in https://github.com/playcanvas/engine/pull/2024